IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/CavityAirflow.mo

```diff
@@ -0,0 +1,105 @@
+within IDEAS.Buildings.Components.BaseClasses.ConvectiveHeatTransfer;
+model CavityAirflow "Model for air flow through a cavity"
+  parameter Boolean linearise = false
+    "=true, to linearise the relation between heat flow rate and temperature difference";
+
+  parameter Real CD=0.65 "Discharge coefficient";
+  parameter Modelica.SIunits.Length h = 2
+    "Height of (rectangular) cavity in wall";
+  parameter Modelica.SIunits.Length w = 1
+    "Width of (rectangular) cavity in wall";
+  parameter Modelica.SIunits.Acceleration g = Modelica.Constants.g_n
+    "Gravity, for computation of buoyancy";
+  parameter Modelica.SIunits.Pressure p = 101300
+    "Absolute pressure for computation of buoyancy";
+  parameter Modelica.SIunits.Density rho = p/r/T
+    "Nominal density for computation of buoyancy mass flow rate";
+  parameter Modelica.SIunits.SpecificHeatCapacity c_p = 1013
+   "Nominal heat capacity for computation of buoyancy heat flow rate";
+  parameter Modelica.SIunits.Temperature T = 293
+   "Nominal temperature for linearising heat flow rate";
+  parameter Modelica.SIunits.TemperatureDifference dT = 1
+   "Nominal temperature difference when linearising heat flow rate"
+   annotation(Dialog(enable=linearise));
+
+  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a port_a
+    "Port for connections between layers"
+    annotation (Placement(transformation(extent={{-110,-10},{-90,10}})));
+  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_b port_b
+    "Port for connections between layers"
+    annotation (Placement(transformation(extent={{90,-10},{110,10}})));
+protected
+  constant Real r = 287 "Gas constant";
+  // Assuming the same entering and leaving flow rate
+  // between .bottom and top half of the cavity
+  // Assuming uniform flow through both parts.
+  // dp= rho * g * h/4 * (1-T1/T2) - buoyancy: horizontal pressure differences
+  // factor 4: distance between door center and center of top/bottom door half
```

Comment:
@Mathadon distance between door center and center top/bottom is h/2. I guess another factor 1/2 is coming from Bernouilli?

I've tried to do the derivation but I'm getting a bit lost and as your implementation is coinciding with the Buildings library, I'll not put more time in it. If you have some spare time, it would be interesting to add a bit of documentation about the derivations but it is not a must.

Reply:
@damienpicard If the door is height h, then the door center is at h/2. And the center of the top/bottom half of the door is at h/4 and 3h/4. So the difference between the former and the latter is h/4 :)